addons/hr/models/res_users.py

```diff
@@ -148,6 +148,7 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
         profile_view = self.env.ref("hr.res_users_view_form_profile")
         if profile_view and view_id == profile_view.id:
             self = self.with_user(SUPERUSER_ID)
+            toolbar = False
         return super(User, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
```

Comment:
As discussed on discord this is what I propose to fix the issue, to avoid removing the whole toolbar instead of just removing the problematic actions.

```suggestion
        profile_view = self.env.ref("hr.res_users_view_form_profile")
        original_user = self.env.user
        if profile_view and view_id == profile_view.id:
            self = self.with_user(SUPERUSER_ID)
        result = super(User, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
        # Due to using the SUPERUSER the result will contain action that the user may not have access too
        # here we filter out actions that requires special implicit rights to avoid having unusable actions
        # in the dropdown menu.
        if toolbar and self.env.user != original_user:
            self = self.with_user(original_user.id)
            if not self.user_has_groups("base.group_erp_manager"):
                change_password_action = self.env.ref("base.change_password_wizard_action")
                action_idx = next(idx for idx, val in enumerate(result['toolbar']['action'])\
                                    if val['id'] == change_password_action.id)
                result['toolbar']['action'].pop(action_idx)
        return result
```